src/Microsoft.AspNet.Mvc.Core/ModelBinding/CollectionModelBinder.cs

```diff
@@ -31,16 +31,19 @@ public class CollectionModelBinder<TElement> : IModelBinder
             var valueProviderResult = await bindingContext.ValueProvider.GetValueAsync(bindingContext.ModelName);
 
             IEnumerable<TElement> boundCollection;
+            CollectionResult result = null;
```

Comment:
i think compiler barfed that i was using an uninitialized value . .. this stays as is,